homeassistant/components/mqtt_room/sensor.py

```diff
@@ -82,28 +88,34 @@ async def async_setup_platform(
 class MQTTRoomSensor(SensorEntity):
     """Representation of a room sensor that is updated via MQTT."""
 
-    def __init__(self, name, state_topic, device_id, timeout, consider_home):
+    def __init__(
+        self, name, state_topic, device_id, timeout, consider_home, extra_rooms
+    ):
         """Initialize the sensor."""
-        self._state = STATE_NOT_HOME
         self._name = name
         self._state_topic = f"{state_topic}/+"
         self._device_id = slugify(device_id).upper()
         self._timeout = timeout
         self._consider_home = (
             timedelta(seconds=consider_home) if consider_home else None
         )
-        self._distance = None
-        self._updated = None
+        self._records = [
+            _create_record(STATE_NOT_HOME, None, None),
+        ]
+
+        for _ in range(extra_rooms):
+            self._records.append(_create_record(STATE_NOT_HOME, None, None))
 
-    async def async_added_to_hass(self) -> None:
+    async def async_added_to_hass(self):
         """Subscribe to MQTT events."""
 
         @callback
-        def update_state(device_id, room, distance):
+        def update_state(record, device_id, room, distance):
```

Comment:
If you use a TypedDict for holding a state you can update all values at once.

Reply:
I'm not 100% sure what you are suggesting, but let me update the code and let me know if that is in the right direction.